SYCL/ESIMD/lsc/lsc_argument_type_deduction.cpp

```diff
@@ -0,0 +1,114 @@
+//==-------- lsc_argument_type_deduction.cpp - DPC++ ESIMD on-device test --==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu-intel-pvc || esimd_emulator
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+// The test checks that compiler is able to corrctly deduce the template
+// parameter types for lsc functions.
+
+#include "../esimd_test_utils.hpp"
+
+#include <algorithm>
+#include <cmath>
+#include <numeric>
```

Comment:
minor comment: <numeric> is needed for std::iota. Other two includes seem redundant.